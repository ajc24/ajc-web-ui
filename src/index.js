/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
