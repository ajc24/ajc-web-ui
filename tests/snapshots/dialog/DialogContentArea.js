import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { DialogContentArea } from '../../../src';

describe('DialogContentArea', () => {
  describe('Default props and rendering - Component with white background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DialogContentArea id="test-dialog-content-area-id">
            Test dialog content area text.
          </DialogContentArea>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with grey background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DialogContentArea id="test-dialog-content-area-id" colour="grey">
            Test dialog content area text.
          </DialogContentArea>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with yellow background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DialogContentArea id="test-dialog-content-area-id" colour="yellow">
            Test dialog content area text.
          </DialogContentArea>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
