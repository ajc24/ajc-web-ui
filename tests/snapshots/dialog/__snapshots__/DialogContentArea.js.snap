// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogContentArea Default props and rendering - Component with white background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-white"
  id="test-dialog-content-area-id-root"
>
  <div
    className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
    id="test-dialog-content-area-id"
  >
    Test dialog content area text.
  </div>
</div>
`;

exports[`DialogContentArea Transferred props and rendering - Component with grey background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-grey-1"
  id="test-dialog-content-area-id-root"
>
  <div
    className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
    id="test-dialog-content-area-id"
  >
    Test dialog content area text.
  </div>
</div>
`;

exports[`DialogContentArea Transferred props and rendering - Component with yellow background verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-left ajc-dialog-content ajc-width-dialog-content ajc-background-yellow"
  id="test-dialog-content-area-id-root"
>
  <div
    className="ajc-flex-column ajc-flex-justify-left ajc-text-black ajc-font-default"
    id="test-dialog-content-area-id"
  >
    Test dialog content area text.
  </div>
</div>
`;
